### PR TITLE
build asteroidsyncservice for starfish and telescope

### PR DESCRIPTION
**Pull request by StefWe**
first try to make it possible with build configuration to build for starfish or telescope

### CODE REVIEW

File: asteroidsyncserviced/asteroidsyncserviced.pro [hunk 1]
```diff
@@ -3,8 +3,9 @@ QT -= gui
 
 WATCHFISH_FEATURES = notificationmonitor walltime music
 
+SUBDIRS += libwatchfish
```
Comment (FlorentRevest) [inline asteroidsyncserviced/asteroidsyncserviced.pro]: Shouldn't this be put inside starfish.pri ?

File: asteroidsyncserviced/telescope.pri [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+SOURCES += platforms/ubuntutouch/main.cpp \
+    watchesmanager.cpp \
```
Comment (FlorentRevest) [inline asteroidsyncserviced/telescope.pri]: Could you keep the shared source files between telescope and starfish in the SOURCES and HEADERS variables of the .pro file? And only keep SOURCES += for the platform specific files?

I think you can also factorize some other variables such as INSTALLS etc...

File: README.md [hunk 3]
```diff
@@ -2,3 +2,12 @@
 A synchronization daemon for [AsteroidOS](http://asteroidos.org/) watches. Forked from [Rockpool](https://github.com/abranson/rockpool).
 
 asteroidsyncserviced is a daemon relying on libasteroid to synchronize data from a given platform to a watch. It comes with a QML module named asteroidsyncservice. It is used by starfish, a SailfishOS synchronization app and Telescope, an Ubuntu Touch synchronization app.
+
+## Building
+
+To build asteroidsyncservice, all submodules must be loaded.  
+   `git submodule update --init`
```
Comment (FlorentRevest) [inline README.md:9]: Please rebase this on the current master branch (which already includes a commit modifying the README with this) for your PR to be mergeable

Comment (StefWe) [reply README.md:9]: Done.

File: asteroidsyncserviced/asteroidsyncserviced.pro (hunk 1 above)
Comment (StefWe) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: done

**Review by StefWe — COMMENTED** [review]
Thank you so much for the tips. Tried to implement that

File: asteroidsyncserviced/platforms/ubuntutouch/main.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#include <QCoreApplication>
+#include <QDebug>
+#include "../../dbusinterface.h"
+#include "../../watchesmanager.h"
+#include "ubuntuplatform.h"
```
Comment (FlorentRevest) [inline asteroidsyncserviced/platforms/ubuntutouch/main.cpp]: Could you keep a shared main.cpp that has its #include "xxplatform.h" and object creation inside an ifdef block?

File: asteroidsyncserviced/asteroidsyncserviced.pro (hunk 1 above)
Comment (FlorentRevest) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: What about "SUBDIRS += libwatchfish" ?

Comment (StefWe) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: If i only include "libwatchfish.pri" it is not possible for me to build the projekt.
In the .pro file we have "QT += dbus" and without that i can not compile it.

File: asteroidsyncserviced/telescope.pri (hunk 2 above)
Comment (StefWe) [reply asteroidsyncserviced/telescope.pri]: done

File: asteroidsyncserviced/asteroidsyncserviced.pro (hunk 1 above)
Comment (FlorentRevest) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: But what about putting the "SUBDIRS += libwatchfish" line inside starfish.pri ?

Comment (StefWe) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: Is libwatch only needed by starfish? If yes, than i put it inside starfish.pri. For sure...

Comment (FlorentRevest) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: Afaik libwatchfish is mostly sailfishos-specific and wouldn't work on UBPorts.

Comment (StefWe) [reply asteroidsyncserviced/asteroidsyncserviced.pro]: Thank you so much for the explanation. I have not yet understood and looked at all this so deeply.

File: asteroidsyncserviced/platforms/ubuntutouch/main.cpp (hunk 4 above)
Comment (StefWe) [reply asteroidsyncserviced/platforms/ubuntutouch/main.cpp]: I have to see if I can do this

Comment (StefWe) [reply asteroidsyncserviced/platforms/ubuntutouch/main.cpp]: I made a first try with ifdef block defined in the *.pri files
```suggestion
    #if UBUNTU_TOUCH_PLATFORM
      UbuntuPlatform *platform = new UbuntuPlatform(watchesManager);
    #endif    
    #if SAILFISHOS_PLATFORM
      SailfishPlatform *platform = new SailfishPlatform(watchesManager);
    #endif
```

**Pull request merged**